tests/ptf/ptf_runner.py

```diff
@@ -197,21 +200,33 @@ def run_test(p4info_path, grpc_addr, device_id, cpu_port, ptfdir, port_map_path,
     # "ptf_port" is ignored for now, we assume that ports are provided by
     # increasing values of ptf_port, in the range [0, NUM_IFACES[.
     port_map = OrderedDict()
+    interfaces = ""
     with open(port_map_path, 'r') as port_map_f:
         port_list = json.load(port_map_f)
-        # Create new portmap proto object for testvectors
-        tv_portmap = pmutils.get_new_portmap()
+        if generate_tv:
+            # Create new portmap proto object for testvectors
+            tv_portmap = pmutils.get_new_portmap()
         for entry in port_list:
             p4_port = entry["p4_port"]
             iface_name = entry["iface_name"]
             port_map[p4_port] = iface_name
-            # Append new entry to tv proto object
-            pmutils.add_new_entry(tv_portmap, p4_port, iface_name)
-
-    # Write the portmap proto object to testvectors/portmap.pb.txt
-    pmutils.write_to_file(tv_portmap, os.getcwd())
+            if generate_tv:
+                # Append new entry to tv proto object
+                interfaces = interfaces + " " + iface_name
+                pmutils.add_new_entry(tv_portmap, p4_port, iface_name)
+    if generate_tv:
+        try:
+            cmd = os.getcwd() + "/../../run/tv/setup_interfaces.sh" + interfaces
```

Comment:
Please add a comment to explain why you're setting up interfaces

Reply:
Done.